src/java.base/share/classes/sun/net/www/protocol/https/HttpsClient.java

```diff
@@ -450,11 +451,6 @@ public void afterConnect() throws IOException, UnknownHostException {
                 if (!(serverSocket instanceof SSLSocket)) {
                     s = (SSLSocket)factory.createSocket(serverSocket,
                                                         host, port, true);
-                } else {
-                    s = (SSLSocket)serverSocket;
```

Comment:
This line should not have been deleted - otherwise NPE will arise later on - e.g. at line 474. Has this change really been tested?

Reply:
right,  i just saw failure(66)  in my local environment. I wanted to delete only the below two lines.
if (s instanceof SSLSocketImpl) {
                        ((SSLSocketImpl)s).setHost(host);
                    }
I will update the PR soon, once my  local test successful.

Reply:
Hi Daniel, i updated the PR with your suggested change.